File: backend/src/main/java/es/dimecresalessis/scoutbase/infrastructure/player/web/dto/PlayerCreateRequest.java
package es.dimecresalessis.scoutbase.infrastructure.player.web.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import es.dimecresalessis.scoutbase.domain.player.model.Player;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Request DTO for creating a new {@link Player} entity.
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class PlayerCreateRequest {

    @NotBlank
    private String name;

    @NotBlank
    private String surname;

    private int birthYear;

    private String email;

    private int number;

    private String position;

    private int priority;
}

File: mobileFrontend/app/src/main/java/com/empresa/scoutbase/model/player/PlayerCreateRequest.kt
// com/empresa/scoutbase/model/player/PlayerCreateRequest.kt
package com.empresa.scoutbase.model.player

data class PlayerCreateRequest(
    val name: String,
    val surname: String,
    val birthYear: Int,
    val email: String,
    val number: Int,
    val position: String,
    val priority: Int
)


